UI fixes concerning when there is no "something".

Thimphou opened the pull request:
## Objective
This PR aims to improve the app's design when: 
- There is no activities found. 
- The user has no friend requests. 

## Modifications

### DiscoverScreen

Before, there was nohting showed on screen when no activities were fetched, now the user knows what is going on as you can see below! 

<img src="https://github.com/LASTA-OUTDOOR/lasta/assets/91149580/66906bac-4eaa-4312-b8bf-b0bb08928f1b" width="200"/>


### FriendsRequestList

Improved UI of this page. 

| Before                                      | After                                       |
|---------------------------------------------|---------------------------------------------|
| <img src="https://github.com/LASTA-OUTDOOR/lasta/assets/91149580/fe67700b-b753-4a42-a9bf-ba55f53c5378" width="200"/> | <img src="https://github.com/LASTA-OUTDOOR/lasta/assets/91149580/9861e229-2493-46fd-b0cc-88a163aa3dd7" width="200"/> |

### MoreInfoScreen

Quick fix concerning the basic icons of activities (previously it was markers, now it's only what's inside the markers), but it is very straightforward.

Thimphou committed:
No activities found

Thimphou committed:
Added icon

Thimphou committed:
advanced

Thimphou committed:
Formatted

Thimphou committed:
Adapted test

Thimphou committed:
Merge branch 'main' into discover/noActivities

Thimphou committed:
Added no_friends resource

Thimphou committed:
Added tests for empty list of activities

Thimphou committed:
Added tests for MoreInfoScreen icons

Thimphou committed:
Formatted

JoniDutoit commented on app/src/main/java/com/lastaoutdoor/lasta/ui/screen/discover/DiscoverScreen.kt:
```diff
@@ -115,7 +119,7 @@ fun DiscoverScreen(
             LoadingAnim(width = 35, tag = "LoadingBarDiscover")
           } else if (discoverScreenState.activities.isEmpty()) {
-            /* TODO */
+            EmptyActivityList()
```
Nice! This Todo was annoying me hahah

JoniDutoit commented on app/src/main/java/com/lastaoutdoor/lasta/ui/screen/discover/DiscoverScreen.kt:
```diff
@@ -445,3 +449,22 @@ fun ActivitiesDisplay(
+            text = LocalContext.current.getString(R.string.no_activities),
+            style =
+                TextStyle(
+                    fontWeight = FontWeight.Normal, fontSize = 20.sp, textAlign = TextAlign.Center),
```
For the text, is better to use MaterialTheme.typography package for different sort of texts so that it looks homegenous through out the app! Hardcoded text fonts wont adapted themselves with the screen size which is an UI issue

JoniDutoit commented on app/src/main/java/com/lastaoutdoor/lasta/ui/screen/moreinfo/MoreInfoScreen.kt:
```diff
@@ -456,7 +462,7 @@ fun ActivityPicture(activityToDisplay: Activity) {
           painter = painterResource(id = defaultId),
           contentDescription = "Default Activity Picture",
-          modifier = Modifier.padding(5.dp).size(90.dp))
+          modifier = Modifier.padding(5.dp).size(90.dp).testTag(testTag))
```
The activity icon looks rather big on my screen, did you try with smaller sizes to see how it goes ?

JoniDutoit commented on app/src/main/java/com/lastaoutdoor/lasta/ui/screen/social/components/FriendsRequestList.kt:
```diff
@@ -41,18 +47,38 @@ fun FriendsRequestList(
+            Text(
+                text = LocalContext.current.getString(R.string.no_friends_yet),
+                style =
+                    TextStyle(
```
Here again the font is unfortunately hard coded with size 20.sp and wont adapted to the screen size, consider changing it to a font from the package MaterialTheme.typography provided by Android

JoniDutoit reviewed (changes requested):
Amazing work! Really like the icons you added, also good idea changing the icon from the MoreInfoScreen to the rounded one instead! It does indeed add information for a better UX experience. I left some suggestions so that you could improve some aspects of your implementation, otherwise it looks good to me!

Thimphou committed:
Changes made + Fixing UI of two other pages

Thimphou committed:
Changed size

Thimphou commented:
Thanks @JoniDutoit for these useful remarks! I changed the size of the icons on the MoreInfoScreen, adapted the TextStyle previously used into MaterialTheme, and I also added an Icon on Feed and Friends tabs on Social Screen, and modularized it!

Hope these changes will make you happy :)

Thimphou committed:
Merge remote-tracking branch 'origin' into discover/noActivities
